Remove more MSVC hacks

Constellation opened the pull request:
#### b1897a24b1f14d6acd8a60244d8ee5e09f31f4ae
<pre>
Remove more MSVC hacks
<a href="https://bugs.webkit.org/show_bug.cgi?id=274018">https://bugs.webkit.org/show_bug.cgi?id=274018</a>
<a href="https://rdar.apple.com/127899955">rdar://127899955</a>

Reviewed by Mark Lam.

* Source/JavaScriptCore/jit/JITMathIC.h:
(JSC::JITMathIC::generateOutOfLine):
* Source/JavaScriptCore/jit/JITOperations.cpp:
* Source/JavaScriptCore/llint/LLIntData.h:
(JSC::LLInt::getCodeFunctionPtr):
(JSC::LLInt::getWide16CodeFunctionPtr):
(JSC::LLInt::getWide32CodeFunctionPtr):
* Source/JavaScriptCore/llint/LLIntThunks.cpp:
(JSC::LLInt::untaggedPtr):
* Source/JavaScriptCore/runtime/CachedTypes.cpp:
(JSC::GenericCacheEntry::decode const):
* Source/WTF/wtf/Assertions.h:
* Source/WTF/wtf/MathExtras.h:
(WTF::clz):
(WTF::ctz):
(wtf_pow): Deleted.
* Source/WTF/wtf/ThreadSafeWeakPtr.h:
* Source/WTF/wtf/TypeTraits.h:
(WTF::requires):
* Source/WebCore/Modules/async-clipboard/ClipboardItemBindingsDataSource.cpp:
(WebCore::ClipboardItemBindingsDataSource::collectDataForWriting):
* Source/WebCore/fileapi/AsyncFileStream.cpp:
(WebCore::AsyncFileStream::Internals::Internals):
* Source/WebCore/loader/SubresourceLoader.h:
* Source/WebCore/platform/PODInterval.h:
* Source/WebCore/platform/PlatformMouseEvent.h:
(WebCore::operator==): Deleted.
(WebCore::operator!=): Deleted.
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.h:
* Source/WebCore/svg/properties/SVGAnimatedPropertyList.h:
* Source/WebCore/svg/properties/SVGProperty.h:
* Source/WebCore/svg/properties/SVGPropertyList.h:
* Source/WebCore/svg/properties/SVGValueProperty.h:
* Source/WebCore/svg/properties/SVGValuePropertyList.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didFailToSuspendAfterProcessSwap):
(WebKit::WebPageProxy::didSuspendAfterProcessSwap):

Canonical link: <a href="https://commits.webkit.org/278643@main">https://commits.webkit.org/278643@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7e12e39baec868882a82952dd457d917ed219098

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51204 "Failed to checkout and rebase branch from PR 28410") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30506 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3539 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/54461 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/1894 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/53507 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36803 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1568 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/54461 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53303 "Failed to checkout and rebase branch from PR 28410") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/28134 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/44137 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/54461 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/25454 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/1393 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/44543 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/47431 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/1466 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56057 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/50706 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/26314 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/1352 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/49096 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/27562 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/44204 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/48238 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/28443 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/62944 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7444 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/27293 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/62944 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/JavaScriptCore/runtime/CachedTypes.cpp:
```diff
@@ -2543,10 +2543,8 @@ bool GenericCacheEntry::decode(Decoder& decoder, std::pair<SourceCodeKey, Unlink
     }
     RELEASE_ASSERT_NOT_REACHED();
-#if COMPILER(MSVC)
     // Without this, MSVC will complain that this path does not return a value.
```
Let's remove this comment too since MSVC is no longer a thing.

MenloDorian reviewed (approved):
r=me with MSVC comments removed also, and if EWS is green.

Constellation committed:
Remove more MSVC hacks
https://bugs.webkit.org/show_bug.cgi?id=274018
rdar://127899955

Reviewed by Mark Lam.

* Source/JavaScriptCore/jit/JITMathIC.h:
(JSC::JITMathIC::generateOutOfLine):
* Source/JavaScriptCore/jit/JITOperations.cpp:
* Source/JavaScriptCore/llint/LLIntData.h:
(JSC::LLInt::getCodeFunctionPtr):
(JSC::LLInt::getWide16CodeFunctionPtr):
(JSC::LLInt::getWide32CodeFunctionPtr):
* Source/JavaScriptCore/llint/LLIntThunks.cpp:
(JSC::LLInt::untaggedPtr):
* Source/JavaScriptCore/runtime/CachedTypes.cpp:
(JSC::GenericCacheEntry::decode const):
* Source/WTF/wtf/Assertions.h:
* Source/WTF/wtf/MathExtras.h:
(WTF::clz):
(WTF::ctz):
(wtf_pow): Deleted.
* Source/WTF/wtf/ThreadSafeWeakPtr.h:
* Source/WTF/wtf/TypeTraits.h:
(WTF::requires):
* Source/WebCore/Modules/async-clipboard/ClipboardItemBindingsDataSource.cpp:
(WebCore::ClipboardItemBindingsDataSource::collectDataForWriting):
* Source/WebCore/fileapi/AsyncFileStream.cpp:
(WebCore::AsyncFileStream::Internals::Internals):
* Source/WebCore/loader/SubresourceLoader.h:
* Source/WebCore/platform/PODInterval.h:
* Source/WebCore/platform/PlatformMouseEvent.h:
(WebCore::operator==): Deleted.
(WebCore::operator!=): Deleted.
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.h:
* Source/WebCore/svg/properties/SVGAnimatedPropertyList.h:
* Source/WebCore/svg/properties/SVGProperty.h:
* Source/WebCore/svg/properties/SVGPropertyList.h:
* Source/WebCore/svg/properties/SVGValueProperty.h:
* Source/WebCore/svg/properties/SVGValuePropertyList.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didFailToSuspendAfterProcessSwap):
(WebKit::WebPageProxy::didSuspendAfterProcessSwap):

Canonical link: https://commits.webkit.org/278643@main
